cmd/check_container.go

```diff
@@ -66,41 +76,25 @@ var checkContainerCmd = &cobra.Command{
 		}
 
 		projectId := viper.GetString("certification_project_id")
-		if projectId == "" {
-			return errors.ErrEmptyProjectID
-		}
-		if strings.HasPrefix(projectId, "ospid-") {
-			projectId = strings.Split(projectId, "-")[1]
-			// Since we want the modified version, write it back
-			// to viper so that subsequent calls don't need to check
-			viper.Set("certification_project_id", projectId)
-		}
-		apiToken := viper.GetString("pyxis_api_token")
-		pyxisClient := pyxis.NewPyxisClient(viper.GetString("pyxis_host"), apiToken, projectId, &http.Client{Timeout: 60 * time.Second})
-		certProject, err := pyxisClient.GetProject(ctx)
-		if err != nil {
-			log.Error(err, "could not retrieve project")
-			return err
-		}
-		log.Debugf("Certification project name is: %s", certProject.Name)
-		if certProject.Container.OsContentType == "scratch" {
-			cfg.EnabledChecks = engine.ScratchContainerPolicy()
-			cfg.Scratch = true
-		}
-
-		if submit {
-			dockerConfigJsonFile, err := os.Open(viper.GetString("dockerConfig"))
-			if err != nil {
-				return err
+		if projectId != "" {
+			if strings.HasPrefix(projectId, "ospid-") {
+				projectId = strings.Split(projectId, "-")[1]
+				// Since we want the modified version, write it back
+				// to viper so that subsequent calls don't need to check
+				viper.Set("certification_project_id", projectId)
 			}
-			defer dockerConfigJsonFile.Close()
-
-			dockerConfigJsonBytes, err := io.ReadAll(dockerConfigJsonFile)
+			apiToken := viper.GetString("pyxis_api_token")
+			pyxisClient := pyxis.NewPyxisClient(viper.GetString("pyxis_host"), apiToken, projectId, &http.Client{Timeout: 60 * time.Second})
+			certProject, err := pyxisClient.GetProject(ctx)
```

Comment:
Does this also need to be wrapped in an `if submit { .. } ` block?

Reply:
No. This is by design. If you don't supply a projectId, it won't run. The only reason this block is running, is to check if it is a scratch container. So it doesn't have anything to do with submit.

Reply:
Might be an edge case, but if a user provides a project ID (for whatever reason), but doesn't pass the submit flag, we'd hit this block. If that same user provided nothing else (as in, no api token, docker config, etc., because they didn't want to submit), we'd hit error cases here (unable to hit Pyxis because the user's token wasn't provided). Am I misreading? Edge case too small?

I'm good either way.

Reply:
That would happen, yes. It just means that GetProject will error out with a 401 when the API token is empty.
I could make it so that if certification-project-id is supplied, then api-token *must* be supplied. Then it would be caught before even getting here. As for docker config, that's moving to within the if submit at the end of the command.

Reply:
I don't think there is a way for `cobra` to say `flag-a` depends on `flag-b` but if that's a thing that would be nice to add upfront for this edge case.

Reply:
Well, sort of. You have to do it yourself. Look at the Args section of the cmd. We're already doing it with --submit and docker-config.

Reply:
So this lets us check our container against scratch checks without requiring submit. Okay, I think that's a better benefit than the edge error case. LGTM

Reply:
So this lets us check our container against scratch checks without requiring submit. Okay, I think that's a better benefit than the edge error case. LGTM